app/src/main/java/org/desarrolladorslp/teckersapp/MainActivity.kt

```diff
@@ -99,31 +126,49 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
         }
     }
 
-    private fun firebaseAuthWithGoogle(acct: GoogleSignInAccount) {
-        val credential = GoogleAuthProvider.getCredential(acct.idToken, null)
+    private fun firebaseAuthWithGoogle(googleSignInAccount: GoogleSignInAccount) {
+        val credential = GoogleAuthProvider.getCredential(googleSignInAccount.idToken, null)
         auth.signInWithCredential(credential)
             .addOnCompleteListener(this) { task ->
                 if (task.isSuccessful) {
                     val user = auth.currentUser
-                    setProfileInformation(acct)
-                    updateUI(user)
+                    val firebaseTokenId = user?.zze()?.zzd()
+
+                    setProfileInformation(googleSignInAccount, firebaseTokenId)
                 } else {
-                    Snackbar.make(main_layout, "Authentication Failed.", Snackbar.LENGTH_SHORT).show()
+                    Snackbar.make(main_layout, "Authentication Failed.", Snackbar.LENGTH_SHORT)
+                        .show()
                     updateUI(null)
                 }
 
             }
     }
 
-    private fun setProfileInformation(acct: GoogleSignInAccount?){
+    private fun setProfileInformation(googleSignInAccount: GoogleSignInAccount, firebaseTokenId: String?) {
+
+        if (googleSignInAccount != null) {
+            var loginService = RetrofitManager.instance()?.create(LoginService::class.java);
+
+            var loginCall = loginService?.login("firebase", firebaseTokenId)
 
-        if (acct != null) {
-            user = User(acct)
+            loginCall?.enqueue(object : Callback<LoggedUser> {
+                override fun onResponse(call: Call<LoggedUser>, response: Response<LoggedUser>) {
+                    APIEndpoint.setAccessToken(response.body()?.accessToken)
+                    user = User(googleSignInAccount)
+                    updateUI(auth.currentUser)
+                }
+
+                override fun onFailure(call: Call<LoggedUser>, t: Throwable) {
+                    if(t is AuthorizationException)
```

Comment:
Maybe any error on the login request should do the same: redirect the user to login page show an error and prevent  the user from enter the application. Do you agree? (I mean, no need to validate the exception here, always call to `authorizationFailure()`)

Reply:
This is the login page